src/libopensc/pkcs15-prkey.c

```diff
@@ -569,20 +569,12 @@ sc_pkcs15_erase_prkey(struct sc_pkcs15_prkey *key)
 	case SC_ALGORITHM_GOSTR3410:
 		free(key->u.gostr3410.d.data);
 		break;
-	case SC_ALGORITHM_EC:
+	case SC_ALGORITHM_EC: /* EC, Edwards and Montgomery use common ec params */
 		free(key->u.ec.params.der.value);
 		free(key->u.ec.params.named_curve);
 		free(key->u.ec.privateD.data);
 		free(key->u.ec.ecpointQ.value);
```

Comment:
should we set them to null to avoid UAF or we will make sure its not used anywhere after this?

Reply:
Not needed as line: 
`d1acf3bab9 (Frank Morgner       2023-09-15 13:07:05 +0200 579)  sc_mem_clear(key, sizeof(*key));`
clears the key for all types of keys.